### PR TITLE
TGP-2300: Implement send email

### PR DESCRIPTION
**Pull request by yohan-oc**

### CODE REVIEW

File: app/uk/gov/hmrc/tradergoodsprofilesdatastore/controllers/DownloadDataSummaryController.scala [hunk 1]
```diff
@@ -78,10 +81,26 @@ class DownloadDataSummaryController @Inject() (
...
+          customsDataStoreConnector.getEmail(request.eori).flatMap {
+            case Some(email) =>
+              emailConnector
+                .sendDownloadRecordEmail(email.address, DownloadRecordEmailParameters(retentionDays))
+                .map { _ =>
+                  NoContent
+                }
+                .recover { case ex =>
+                  logger.error(s"Failed to send download record email: ${ex.getMessage}")
+                  BadRequest(s"Failed to send download record email")
```
Comment (besscerule) [inline app/uk/gov/hmrc/tradergoodsprofilesdatastore/controllers/DownloadDataSummaryController.scala:96]: I would try to have the status code from the ex in the logging

File: conf/application.conf [hunk 2]
```diff
@@ -92,6 +92,12 @@ microservice {
       host     = localhost
       port     = 9025
     }
+
+    email {
```
Comment (besscerule) [inline conf/application.conf:96]: do we need changes to the service profiles or the app.config for different environments? @ChrisCookOC is doing the testing spike so might be able to help with this

File: it/test/uk/gov/hmrc/tradergoodsprofilesdatastore/connectors/EmailConnectorSpec.scala [hunk 3]
```diff
@@ -0,0 +1,90 @@
...
+  implicit private lazy val hc: HeaderCarrier = HeaderCarrier()
+
+  ".sendDownloadRecordEmail" - {
+
+    val testEmail                         = "someone@somewher.com"
+    val testExpiredDate                   = "05 Sept 2024"
+    val testDownloadRecordEmailParameters = DownloadRecordEmailParameters(testExpiredDate)
+    val testDownloadRecordEmailRequest    = DownloadRecordEmailRequest(Seq(testEmail), testDownloadRecordEmailParameters)
+
+    "must send email2" in {
```
Comment (besscerule) [inline it/test/uk/gov/hmrc/tradergoodsprofilesdatastore/connectors/EmailConnectorSpec.scala:61]: typo?

File: app/uk/gov/hmrc/tradergoodsprofilesdatastore/controllers/DownloadDataSummaryController.scala [hunk 4]
```diff
@@ -78,10 +81,26 @@ class DownloadDataSummaryController @Inject() (
...
+                }
+
+            case None =>
+              logger.error(s"Unable to find the email for EORI: ${request.eori}")
+              Future.successful(BadRequest(s"Unable to find the email for EORI: ${request.eori}"))
```
[the remark this answers is not in the document]
Comment (yohan-oc) [reply app/uk/gov/hmrc/tradergoodsprofilesdatastore/controllers/DownloadDataSummaryController.scala]: Updated

File: app/uk/gov/hmrc/tradergoodsprofilesdatastore/controllers/DownloadDataSummaryController.scala (hunk 1 above)
Comment (yohan-oc) [reply app/uk/gov/hmrc/tradergoodsprofilesdatastore/controllers/DownloadDataSummaryController.scala]: This is now removed

File: conf/application.conf (hunk 2 above)
Comment (ChrisCookOC) [reply conf/application.conf:96]: Will need a change on app-config-base to point to the service properly.

May need further changes for testing didn't get that far today :(

Comment (yohan-oc) [reply conf/application.conf:96]: I created a PR for this https://github.com/hmrc/app-config-base/pull/10985

**Comment by besscerule** [comment]
Approved - however i would wait to merge until my 2303 has all been merged in

**Review by besscerule — DISMISSED** [review]

**Review by ChrisCookOC — APPROVED** [review]

**Pull request merged**